yakut/cmd/file_client/_cmd.py

```diff
@@ -0,0 +1,443 @@
+    timeout: float,
+) -> None:
+    """
+    Remove a file or directory on remote node(s) using the standard Cyphal file service.
+    """
+    from pycyphal.application.file import FileClient2
+
+    _logger.debug("node_ids=%r, path=%r, timeout=%r", node_ids, path, timeout)
+    node_ids_list = list(sorted(node_ids)) if isinstance(node_ids, set) else [node_ids]
+    assert isinstance(node_ids_list, list) and all(isinstance(x, int) for x in node_ids_list)
+
+    error = False
+    with purser.get_node("file_client_rm", allow_anonymous=False) as node:
+        prog = ProgressReporter()
+        for nid in node_ids_list:
+            fc = FileClient2(node, nid, response_timeout=timeout)
+            prog(f"Remove from node {nid}")
+
+            try:
+                await fc.remove(str(path))
+                _logger.info("Removed path %r on node %r", path, nid)
+            except FileNotFoundError:
+                show_warning(f"Path {path} not found on node {nid}")
+            except Exception as e:
+                show_error(f"Error removing {path} on node {nid}: {e}")
+                error = True
+
+    return EXIT_CODE_UNSUCCESSFUL if error else 0
+
+
+@file_client.command()
+@click.argument("node_id", type=int)
+@click.argument("src")
+@click.argument("dst", required=False)
+@click.option(
+    "--timeout",
+    "-T",
+    type=float,
+    default=pycyphal.presentation.DEFAULT_SERVICE_REQUEST_TIMEOUT,
+    show_default=True,
+    metavar="SECONDS",
+    help="Service response timeout.",
+)
+@yakut.pass_purser
+@yakut.asynchronous(interrupted_ok=True)
+async def read(
+    purser: yakut.Purser,
+    node_id: int,
+    src: str,
+    dst: str | None,
+    timeout: float,
+) -> None:
+    """
+    Read a file from a remote node using the standard Cyphal file service.
+    """
+    from pycyphal.application.file import FileClient2
+
+    src = PurePosixPath(src)
+    dst = Path(dst) if dst else Path(src.name)
+    error = False
+
+    with purser.get_node("file_client_read", allow_anonymous=False) as node:
+        prog = ProgressReporter()
+
+        def read_progress_cb(bytes_read: int, bytes_total: int | None) -> None:
+            prog(f"Read {bytes_read} bytes")
+
+        try:
+            fc = FileClient2(node, node_id, response_timeout=timeout)
+
+            with open(dst, "wb") as out:
+                res = await fc.read(str(src), progress=read_progress_cb)
+                out.write(res)
+
+            _logger.info("Read %d bytes from %r on node %r to %r", len(res), src, node_id, dst)
+
+        except FileNotFoundError:
+            show_error(f"File {src} not found on node {node_id}")
+            error = True
+        except Exception as e:
+            show_error(f"Error reading {src} from node {node_id}: {e}")
+            error = True
+
+    return EXIT_CODE_UNSUCCESSFUL if error else 0
+
+
+@file_client.command()
+@click.argument("node_id", type=int)
+@click.argument("src")
+@click.argument("dst", required=False)
```

Comment:
Do you think it would be better to require dst and make src optional, defaulting to stdin in binary mode instead? This would mirror the read command.